File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class AnalyticsSummary(BaseModel):
    total_users: int
    total_courses: int
    total_enrollments: int
    total_assignments: int
    total_submissions: int
    total_graded_submissions: int
    pending_assignments: int = 0
    completed_lessons: int = 0
    average_grade: float | None = None

File: frontend/src/types/analytics.ts
export type AnalyticsSummary = {
  total_users: number;
  total_courses: number;
  total_enrollments: number;
  total_assignments: number;
  total_submissions: number;
  total_graded_submissions: number;
  pending_assignments: number;
  completed_lessons: number;
  average_grade: number | null;
};
